lib/src/audio_waveforms.dart

```diff
@@ -40,18 +40,21 @@ class _AudioWaveformsState extends State<AudioWaveforms> {
 
   double _initialOffsetPosition = 0.0;
   double _initialPosition = 0.0;
+  Duration currentlyRecordedDuration = Duration.zero;
 
   @override
   void initState() {
     super.initState();
-    widget.recorderController.addListener(() {
-      if (mounted) setState(() {});
-    });
+    widget.recorderController
+      ..addListener(_updateOnControllerUpdate)
+      ..onCurrentDuration.listen((duration) {
```

Comment:
please store the `StreamSubscription` returned by the `listen` method and cancel it when not required.